src/Uno.Extensions.Core/IDispatcher.cs

```diff
@@ -0,0 +1,6 @@
+﻿namespace Uno.Extensions;
+
+public interface IDispatcher
+{
+	Task<TResult> ExecuteAsync<TResult>(Func<Task<TResult>> actionWithResult);
```

Comment:
You may want to provide a CancellationToken as well.

Reply:
Yep, good idea

Reply:
I'll go ahead and add it today